pkg/controller/clusterstate/api/resource_info.go

```diff
@@ -117,15 +119,7 @@ func (r *Resource) Replace(rr *Resource) *Resource {
 
 //Sub subtracts two Resource objects.
 func (r *Resource) Sub(rr *Resource) *Resource {
-	if rr.LessEqual(r) {
-		r.MilliCPU -= rr.MilliCPU
-		r.Memory -= rr.Memory
-		r.GPU -= rr.GPU
-		return r
-	}
-
-	panic(fmt.Errorf("Resource is not sufficient to do operation: <%v> sub <%v>",
-		r, rr))
```

Comment:
In the NonNegSub() func. can you add log error message for each case where there is a negative number so we can debug this problem in the future?

Reply:
Done